src/NServiceBus.Core/Scheduling/ScheduledTaskHandlingBehavior.cs

```diff
@@ -0,0 +1,26 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using Pipeline;
+
+    class ScheduledTaskHandlingBehavior : IBehavior<IIncomingLogicalMessageContext, IIncomingLogicalMessageContext>
+    {
+        public ScheduledTaskHandlingBehavior(DefaultScheduler scheduler)
+        {
+            this.scheduler = scheduler;
+        }
+
+        public Task Invoke(IIncomingLogicalMessageContext context, Func<IIncomingLogicalMessageContext, Task> next)
+        {
+            if (context.Message.Instance is ScheduledTask scheduledTask)
+            {
+                return scheduler.Start(scheduledTask.TaskId, context);
```

Comment:
Shortcutting the pipe is a change in behaviour but I can't see a use case for us triggering things like generic `IHandleMessages<IMessage>` handlers?

Reply:
do you want me to call next but set context.Handled?